### ISSUE TITLE
Error in `create` method results in unhandled rejection 

### ISSUE THREAD

**Issue by MitMaro** [feathersjs/express#19]
### Steps to reproduce

Create a simple application with a single service that throws an error in `create`. Call the endpoint with Curl.

#### Reduced test case

See gist: https://gist.github.com/MitMaro/7f05130ad19325c165bd23d44a8d13eb

```javascript
const feathers = require('@feathersjs/feathers');
const express = require('@feathersjs/express');
const app = express(feathers());
app.configure(express.rest());
app.use('/myservice', {
	async create(data, params) {
		throw new Error();
	},
});
app.listen(3030);
console.log('Feathers app started on 127.0.0.1:3030');

```

### Expected behavior

Throwing an error in `create` should not cause an unhandled rejection error. The service should respond with the error.

### Actual behavior

A unhandled rejection warning is displayed, and no response occurs.

**Console 1**:

```shell
$ node index.js
Feathers app started on 127.0.0.1:3030
(node:17001) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'message' of undefined
(node:17001) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Console 2**:
```shell
$ curl -XPOST 127.0.0.1:3030/myservice
```

By adding a `unhandledRejection` handler:

```javascript
process.on('unhandledRejection', (reason) => {
	console.error('Unhandled Rejection at: Promise');
	console.error(reason);
});
```

The following is returned:

```
Unhandled Rejection at: Promise
TypeError: Cannot read property 'message' of undefined
    at service.(anonymous function).then.catch.hook (/Users/timoram/code/broken-project-test-cases/feathers-express-error/node_modules/@feathersjs/express/lib/rest/wrappers.js:79:46)
    at <anonymous>
    at runMicrotasksCallback (internal/process/next_tick.js:121:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

### System configuration

**Module versions** :
```shell
$ npm list --depth=0
feathers-express-error@1.0.0 /Users/timoram/code/broken-project-test-cases/feathers-express-error
├── @feathersjs/express@1.1.2
└── @feathersjs/feathers@3.1.0
```

**NodeJS version**:
```shell
$ npm --version
5.6.0
$ node --version
v8.9.4
```

**Operating System**:
```shell
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.12.5
BuildVersion:	16F73
```

**Browser Version**: N/A

**React Native Version**: N/A

**Module Loader**: N/A

**Comment by daffl** [comment]
That is indeed a bug. The error that you were probably actually getting is `Error: A data object must be provided to the 'create' method` since there is no body parser.

The problem why it got swallowed was that Feathers argument validation ran outside of the hook chain but the REST handler always expected to get a valid hook object and just got a normal error in that case. The fix in https://github.com/feathersjs/feathers/pull/818 (released as `@feathersjs/feathers@v3.1.1`) was to run argument validation as a normal hook as the very first hook.

**Comment by MitMaro** [comment]
Valid point. In the project that initially triggered the problem I believe an error was being thrown in a before hook, but I guess the result would be similar. Thanks!
